src/device/buttplugIoDeviceProvider.ts

```diff
@@ -0,0 +1,95 @@
+import DeviceProvider from "./deviceProvider.js";
+import {ButtplugClientDevice, ButtplugClient, ButtplugNodeWebsocketClientConnector} from "buttplug"
+import Device from "./device.js";
+import ButtplugIoDeviceFactory from "./buttplugIoDeviceFactory.js";
+import DeviceState from "./deviceState.js";
+import EventEmitter from "events";
+
+
+export default class ButtplugIoDeviceProvider extends DeviceProvider
+{
+    private connectedDevices: Map<number, Device> = new Map();
+    private managedDevices: Map<number, null> = new Map();
+
+    private buttplugConnector:  ButtplugNodeWebsocketClientConnector;
+    private buttplugClient:  ButtplugClient;
+
+
+    private readonly buttplugIoDeviceFactory: ButtplugIoDeviceFactory;
+
+    public constructor(eventEmitter: EventEmitter, deviceFactory: ButtplugIoDeviceFactory) {
+        super(eventEmitter);
+        this.buttplugIoDeviceFactory = deviceFactory;
+    }
+
+    public init(): void
+    {
+        if (process.env.INTIFACE_PORT) {
```

Comment:
I think eventually we should configure the different providers in `settings.json` and only instantiate the provider if it's actually configured there. Will adjust my POC implementation in the coming days. Eventually there should be a user interface under "Settings" menu point to add/edit/remove providers.

Reply:
I didn't know how to do that at first this is why I used the port in env as a way to control that, but I was not really happy with this either

Reply:
My energy level is currently rather low so I don’t spend too much time working on SlvCtrl+ currently. But today I started with the refactoring. I did good progress but it needs another few hours of work until it’s done. Sorry for the delay.